DoxyFiles/man/man3/HillDeboer.3
.TH "HillDeboer" 3 "Segunda, 3 de Outubro de 2022" "Version 1.0.0" "Isotherm++" \" -*- nroff -*-
.ad l
.nh
.SH NAME
HillDeboer \- Classe com as equacoes da isoterma de Hill-Deboer\&.  

.SH SYNOPSIS
.br
.PP
.PP
\fC#include <HillDeBoer\&.h>\fP
.PP
Subclasse de \fBThreeParameters\fP e \fBIsothermTemplate< HillDeboer >\fP\&.
.SS "Membros Públicos"

.in +1c
.ti -1c
.RI "\fBDefineIdentity\fP ('HillDeBoer', ID::HillDeBoer)"
.br
.RI "Definicao de \fBID\fP para esta classe para fins de identificacao de erros\&. "
.ti -1c
.RI "\fBHillDeboer\fP ()=default"
.br
.ti -1c
.RI "\fBHillDeboer\fP (const \fBHillDeboer\fP &_orig)=default"
.br
.RI "Construtora de cópia\&. "
.ti -1c
.RI "virtual \fB~HillDeboer\fP ()=default"
.br
.RI "Destrutora\&. "
.ti -1c
.RI "\fBHillDeboer\fP (const Real &_qmax, const Real &_k1, const Real &_k2, const Real &_rgas=Isotherm::RGASCONST)"
.br
.RI "Construtora com os parametros que definem a isoterma de \fBHillDeboer\fP\&. "
.ti -1c
.RI "\fBHillDeboer\fP & \fBoperator=\fP (const \fBHillDeboer\fP &_orig)=default"
.br
.RI "Sobrecarga do operador =\&. "
.ti -1c
.RI "Real \fBQmax\fP () const"
.br
.RI "Funcao que informa o valor da capacidade maxima de adsorcao\&. "
.ti -1c
.RI "Real \fBK1\fP () const"
.br
.RI "Funcao que retorna o coeficiente associado a constante de equilibrio de \fBHill\fP - Deboer\&. "
.ti -1c
.RI "Real \fBK2\fP () const"
.br
.RI "Funcao que retorna o valor associado a constante de iteracao entre as moleculas adsorvidas\&. "
.ti -1c
.RI "Real \fBRgas\fP () const"
.br
.RI "Funcao que retorna o valor da constante universal dos gases\&. "
.ti -1c
.RI "void \fBQmax\fP (const Real &_qmax)"
.br
.RI "Funcao para alterar a capacidade maxima de adsorcao\&. "
.ti -1c
.RI "void \fBK1\fP (const Real &_k1)"
.br
.RI "Funcao que altera o coeficiente associado a constante de equilibrio de \fBHill\fP – Deboer\&. "
.ti -1c
.RI "void \fBK2\fP (const Real &_k2)"
.br
.RI "Funcao para alterar o valor associado a constante de iteracao entre as moleculas adsorvidas\&. "
.ti -1c
.RI "void \fBRgas\fP (const Real &_rgas)"
.br
.RI "Funcao para alterar o valor da constante universal dos gases\&. "
.ti -1c
.RI "virtual Real \fBQe\fP (const Real &_c, const Real &=0) const"
.br
.RI "Funcao que calcula a quantidade de sorcao no equilibrio\&. "
.ti -1c
.RI "virtual std::unique_ptr< \fBIsotherm\fP > \fBCloneImplementation\fP () const override"
.br
.in -1c
.SS "Outros membros herdados"
.SH "Descrição detalhada"
.PP 
Classe com as equacoes da isoterma de Hill-Deboer\&. 

Isoterma com tres parametros, $ q_{max} $, $ K_1 $ e $ K_2 $, cujas formulas sao as seguintes: \\begin{align} [f(\\theta ) = K_1C-\\frac{\\theta}{1-\\theta}e^{\\frac{\\theta}{1-\\theta}-\\frac{K_2\\theta}{RT}}=0] \\end{align} Um artigo de referencia pode ser encontrado \fCaqui\fP\&. 
.PP
\fBAutores\fP
.RS 4
Lara Botelho Brum 
.PP
Joao Flavio Vieira de Vasconcellos 
.RE
.PP
\fBVersão\fP
.RS 4
1\&.0 
.RE
.PP
\fBData\fP
.RS 4
2022 
.RE
.PP
\fBBug\fP
.RS 4
Nao ha bugs conhecidos\&.
.RE
.PP
.PP
\fBCopyright\fP
.RS 4
GNU Public License\&. 
.RE
.PP

.SH "Construtores e Destrutores"
.PP 
.SS "HillDeboer::HillDeboer ()\fC [default]\fP"

.PP
Construtora default\&. 
.PP
Uso: 
.PP
.nf
HillDeboer  var;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros para esta funcao 
.RE
.PP

.SS "HillDeboer::HillDeboer (const \fBHillDeboer\fP & _orig)\fC [default]\fP"

.PP
Construtora de cópia\&. Uso: 
.PP
.nf
HillDeboer  var;
HillDeboer  var1(var);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo \fBHillDeboer\fP original\&. 
.RE
.PP

.SS "HillDeboer::HillDeboer (const Real & _qmax, const Real & _k1, const Real & _k2, const Real & _rgas = \fCIsotherm::RGASCONST\fP)"

.PP
Construtora com os parametros que definem a isoterma de \fBHillDeboer\fP\&. Uso: 
.PP
.nf
double QMAX(1\&.0);
double K1(1\&.0);
double K2(1\&.0);
HillDeboer  var1(Q1, K1, K2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Capacidade maxima de adsorcao\&. 
.br
\fI_k1\fP Constante de equilibrio de \fBHill\fP – Deboer\&. 
.br
\fI_k2\fP Constante de iteracao entre as moleculas adsorvidas\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.br
\fI_k1\fP <= 0\&. 
.br
\fI_k2\fP <= 0\&. 
.RE
.PP

.SH "Funções membros"
.PP 
.SS "virtual std::unique_ptr< \fBIsotherm\fP > HillDeboer::CloneImplementation () const\fC [inline]\fP, \fC [override]\fP, \fC [virtual]\fP"

.PP
Implementa \fBIsotherm\fP\&.
.SS "HillDeboer::DefineIdentity ('HillDeBoer', ID::HillDeBoer)"

.PP
Definicao de \fBID\fP para esta classe para fins de identificacao de erros\&. 
.SS "Real HillDeboer::K1 () const\fC [inline]\fP"

.PP
Funcao que retorna o coeficiente associado a constante de equilibrio de \fBHill\fP - Deboer\&. Uso: 
.PP
.nf
HillDeboer  var1(QMAX, K1, K2);
double k1 = var1\&.K1();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor do coeficiente associado a constante de equilibrio de \fBHill\fP – Deboer\&. 
.RE
.PP

.SS "void HillDeboer::K1 (const Real & _k1)\fC [inline]\fP"

.PP
Funcao que altera o coeficiente associado a constante de equilibrio de \fBHill\fP – Deboer\&. Uso: 
.PP
.nf
HillDeboer  var1(QMAX, K1, K2);
double k1(2\&.0);
var1\&.K1(k1);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_k1\fP Novo valor do coeficiente associado a constante de equilibrio de \fBHill\fP – Deboer\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_k1\fP <= 0\&. 
.RE
.PP

.SS "Real HillDeboer::K2 () const\fC [inline]\fP"

.PP
Funcao que retorna o valor associado a constante de iteracao entre as moleculas adsorvidas\&. Uso: 
.PP
.nf
HillDeboer  var1(QMAX, K1, K2);
double k2 = var1\&.K2();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor associado a constante de iteracao entre as moleculas adsorvidas\&. 
.RE
.PP

.SS "void HillDeboer::K2 (const Real & _k2)\fC [inline]\fP"

.PP
Funcao para alterar o valor associado a constante de iteracao entre as moleculas adsorvidas\&. Uso: 
.PP
.nf
HillDeboer  var1(QMAX, K1, K2);
double k2(3\&.0);
var1\&.K2(k2);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_k2\fP Novo valor associado a constante de iteracao entre as moleculas adsorvidas\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_k2\fP <= 0\&. 
.RE
.PP

.SS "\fBHillDeboer\fP & HillDeboer::operator= (const \fBHillDeboer\fP & _orig)\fC [default]\fP"

.PP
Sobrecarga do operador =\&. Uso: 
.PP
.nf
HillDeboer  var1(QMAX, K1, K2);
HillDeboer  var2 = var1;

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_orig\fP Variavel do tipo HillDeBoer original\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Cópia de _orig\&. 
.RE
.PP

.SS "Real HillDeboer::Qe (const Real & _c, const Real & _temp = \fC0\fP) const\fC [virtual]\fP"

.PP
Funcao que calcula a quantidade de sorcao no equilibrio\&. Uso: 
.PP
.nf
HillDeboer  var1(QMAX, K1, K2);
double ce(1\&.0);
double qe = var1\&.Qe(ce);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_c\fP Concentracao do soluto\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da quantidade de sorcao no equilibrio\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_c\fP < 0\&. 
.RE
.PP

.PP
Implementa \fBIsotherm\fP\&.
.SS "Real HillDeboer::Qmax () const\fC [inline]\fP"

.PP
Funcao que informa o valor da capacidade maxima de adsorcao\&. Uso: 
.PP
.nf
HillDeBoer  var1(QMAX, K1, K2);
double q1 = var1\&.Qmax();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da capacidade maxima de adsorcao\&. 
.RE
.PP

.SS "void HillDeboer::Qmax (const Real & _qmax)\fC [inline]\fP"

.PP
Funcao para alterar a capacidade maxima de adsorcao\&. Uso: 
.PP
.nf
HillDeboer  var1(QMAX, K1, K2);
double q1(3\&.0);
var1\&.Qmax(q1);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_qmax\fP Novo valor da capacidade maxima de adsorcao\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_qmax\fP <= 0\&. 
.RE
.PP

.SS "Real HillDeboer::Rgas () const\fC [inline]\fP"

.PP
Funcao que retorna o valor da constante universal dos gases\&. Uso: 
.PP
.nf
DubininRadushkevich  var1(q1, k1);
Real rg = var1\&.Rgas();

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI \fP Nao ha parametros\&. 
.RE
.PP
\fBRetorna\fP
.RS 4
Valor da constante universal dos gases\&. 
.RE
.PP

.SS "void HillDeboer::Rgas (const Real & _rgas)\fC [inline]\fP"

.PP
Funcao para alterar o valor da constante universal dos gases\&. Uso: 
.PP
.nf
DubininRadushkevich  var1(q1, k1);
Real rgas(62\&.363598221529);
var1\&.Rgas(rgas);

.fi
.PP
 
.PP
\fBParâmetros\fP
.RS 4
\fI_rgas\fP Novo valor da constante universal dos gases\&. 
.RE
.PP
\fBExceções\fP
.RS 4
\fI_rgas\fP <= 0\&. 
.RE
.PP


.SH "Autor"
.PP 
Gerado automaticamente por Doxygen para Isotherm++ a partir do código-fonte\&.


IsothermLib/src/Isotherm/ThreeParameters/HillDeBoer.cpp
//==============================================================================
// Name        : HillDeboer.cpp
// Author      : Iasmim Barboza Storck
//               Lara Botelho Brum
//               Joao Flavio Vieira de Vasconcellos
// Version     : 1.0
// Description : Classe com as equacoes da isoterma de Hill - Deboer
//
// Copyright   : Copyright (C) <2022>  Joao Flavio Vasconcellos
//                                      (jflavio at iprj.uerj.br)
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, either version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//
//==============================================================================



//==============================================================================
// includes lib c++
//==============================================================================

#include <cmath>                // std::exp
#include <iostream>             // std::cout


//==============================================================================
// includes da lib IsothermLib
//==============================================================================

#include <Error/IsoException.h>
#include <Isotherm/ThreeParameters/HillDeboer.h>
#include <Misc/NewtonRaphson.h>


IST_NAMESPACE_OPEN

//==============================================================================
// Variaveis estaticas
//==============================================================================

VecPairString       detailsHillDeboer
    {
        PairString  ( "Qmax"
                    , "Capacidade maxima de adsorcao")
    ,   PairString  ( "K1"
                    , "Constante de equilibrio de Hill Deboer")
    ,   PairString  ( "K2"
                    , "Constante de iteracao entre as moleculas adsorvidas")
    };

template<>
VecPairString IsothermTemplate < HillDeboer >::infoIsotherm = detailsHillDeboer;


//==============================================================================
// Construtora com dois parametros
//==============================================================================

#undef  __FUNCT__
#define __FUNCT__ "HillDeboer :: HillDeboer (const Real&, const Real&, const Real&)"
HillDeboer :: HillDeboer    (   const Real& _qmax
                            ,   const Real& _k1
                            ,   const Real& _k2
                            ,   const Real& _rgas) 
                            :   ThreeParameters(_qmax, _k1, _k2 )
                            ,   RGAS(_rgas) {



    try {
            if (_qmax <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadQmaxLEZero
                                    );

            if (_k1 <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadK1LEZero
                                    );

            if (_k2 < 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadK2LTZero
                                    );

            if (_rgas <= 0.0)  throw
                    IsoException    (   IST_LOC
                                    ,   className()
                                    ,   BadRGasLEZero
                                    );


    } catch (const IsoException& _isoExcept) {

        std::cout   << _isoExcept 
                    << "\n";
        exit(EXIT_FAILURE);

    }

    setup = true;


    }

//==============================================================================
// Concentracao de equilibrio Qe
//==============================================================================

#undef  __FUNCT__
#define __FUNCT__ "HillDeboer ::  Qe (const Real&, const Real&) const"
Real
HillDeboer ::  Qe   (   const Real&     _ce
                    ,   const Real&     _temp
                    ) const {


    try {

        if (!setup) throw
            IsoException    (   IST_LOC
                            ,   className()
                            ,   BadCoefficient
                            );

        if (_ce < 0.0)  throw
                IsoException    (   IST_LOC
                                ,   className()
                                ,   BadCeLTZero
                                );

        if (_temp <= 0.0)  throw
                IsoException    (   IST_LOC
                                ,   className()
                                ,   BadTempLEZero
                                );


    } catch (const IsoException& _isoExcept) {

        std::cout   << _isoExcept 
                    << "\n";
        exit(EXIT_FAILURE);
    }

    const_cast<Real&>(auxiCe) = _ce * this->K1();
    const_cast<Real&>(tempK)  = 1.0 / (_temp * Rgas());
    
    
auto fp    = std::bind(&HillDeboer::FQe, *this, _1);
auto resul = NewtonRaphson (fp, 0.95);
auto value = resul * Qmax();

    return (value >= ZERO ? value : 0.0);

}


#undef  __FUNCT__
#define __FUNCT__ "Real  HillDeboer ::  FQe (const Real&) const"
Real  
HillDeboer ::  FQe (const Real& _theta) const {


auto       auxi1  = _theta / (1.0 - _theta);
auto       auxiK2 = _theta * this->K2() * tempK;
auto       auxi2  = auxi1 * exp(auxi1 - auxiK2);

    return auxiCe - auxi2;

}

IST_NAMESPACE_CLOSE